google/cloud/bigtable/internal/bigtable_stub_factory.cc

```diff
@@ -54,13 +63,22 @@ std::shared_ptr<BigtableStub> CreateBigtableStubRoundRobin(
 std::shared_ptr<BigtableStub> CreateDecoratedStubs(
     google::cloud::CompletionQueue cq, Options const& options,
     BaseBigtableStubFactory const& base_factory) {
+  auto cq_ptr = std::make_shared<CompletionQueue>(cq);
+  auto refresh_state = std::make_shared<ConnectionRefreshState>(
+      cq_ptr, options.get<bigtable::MinConnectionRefreshOption>(),
+      options.get<bigtable::MaxConnectionRefreshOption>());
   auto auth = google::cloud::internal::CreateAuthenticationStrategy(
       std::move(cq), options);
-  auto child_factory = [base_factory, &auth, options](int id) {
-    auto channel = CreateGrpcChannel(*auth, options, id);
+  auto child_factory = [base_factory, cq_ptr, refresh_state, &auth,
```

Comment:
Maybe you want to centralize the code that handles `if (refresh_state->enabled()`, something like this:

```cc
  auto child_factory = [base_factory, cq_ptr, refresh_state, &auth, options](int id) {
    auto channel = CreateGrpcChannel(cq_ptr, *auth, options, id);
    if (refresh_state->enabled()) ScheduleChannelRefresh(cq, refresh_state, channel);
    return base_factory(channel);
  };
  auto stub = CreateBigtableStubRoundRobin(...);
  if (refresh_state->enabled()) {
    stub = std::make_shared<BigtableChannelRefresh>(...);
  }
```